### DISCUSSION TITLE
Pixel coordinate calculation

### DISCUSSION THREAD

**Discussion by Aquachains** [Chlumsky/msdfgen#145] [Q&A] [answered]
Hello together.
First of all, thank you very much for this awesome project Chlumsky, i read your article and master thesis and tried to understand as much as possible.😅 And thanks for all contributors. 👍 
So i write a 2d engine and draw every char with a rect inside a texture.

if i run this snippet:
```
msdfgen.exe mtsdf -font %1 'l' -scale 1 -translate 1 1 -o %~n1.0.png -size 32 32 -pxrange 2 -printmetrics -testrender render.png 1024 1024
```

i get as a result:
![mtsdf0 0](https://user-images.githubusercontent.com/66897141/151159243-97283646-d3d3-4333-b06a-fec31af72719.png)
```
bounds = 2.75, 0, 5.34375, 24.3125
advance = 8.09375
range = 2
```

I need pixel values and i cant get to the real values by rounding or truncating. 
Here the values i see in Gimp (Position and Size inside the red rectangle):
![Generated_MTSDF_I_Gimp](https://user-images.githubusercontent.com/66897141/151164989-9f5df17b-02e8-4f77-af27-88a53d094f71.PNG)
So the bounds should really be **(3, 0, 4, 26)**. (Depending on the direction of the y-Axis) instead of **(2.75, 0, 5.34375, 24.3125)**
I tried different chars and different pixel ranges, but i cant see a pattern how to calculate those values.

Can someone help me how to work with the given values?

**Comment by Chlumsky** [comment, marked as answer]
The formula for pixel bounds based on shape bounds and other parameters may potentially be something along `round(scale*bound+translate+-0.5*pxRange)`, so:
- L: `round(1*2.75+1-0.5*2) = round(2.75) = 3`
- B: `round(1*0+1-0.5*2) = round(0) = 0`
- R: `round(1*5.34375+1+0.5*2) = round(7.34375) = 7`, therefore `width = 7-3 = 4`
- T: `round(1*24.3125+1+0.5*2) = round(26.3125) = 26`, threfore `height = 26-0 = 26`

This could be wrong though, I didn't really check it.

**Reply by Aquachains** [reply]
Hello Chlumsky, i tested a few chars and it works.
Thank you very much. 👍
